spec/views/posts/user_post_index_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe 'user_posts/#index', type: :feature do
+  before :each do
+    @user = User.first
+    visit user_posts_path(@user.id)
+  end
+  describe 'Users Posts index view' do
+    it 'should show user profile picture' do
+      expect(page).to have_css("img[src*='#{@user.photo}']")
+    end
+    it 'should show user username' do
+      expect(page).to have_content(@user.name)
+    end
+    it 'should show the number of posts the user have written' do
+      expect(page).to have_content("Number of posts: #{@user.post_counter}")
+    end
+    it "should show the post's title" do
+      @user.posts.each { |post| expect(page).to have_content(post.title) }
+    end
+    it "should show the post's body" do
+      @user.posts.each { |post| expect(page).to have_content(post.text) }
+    end
+    it "should show the post's first comment" do
+      first_comment = @user.posts.first
+      expect(page).to have_content(first_comment.text)
+    end
+    it 'should show how many comments a post has' do
+      post = @user.posts.first
+      expect(page).to have_content("Comments: #{post.comments_counter}")
+    end
+    it 'should have button for pagination' do
+      expect(page).to have_button('Pagination')
+    end
+  end
+end
```

Comment:
- [ ] You seem to have forgotten to add the tests for "I can see how many likes a post has." Kindly include it.

![image](https://user-images.githubusercontent.com/86318284/217517419-71396146-b505-40df-951e-90702fef32a3.png)

- [ ] You seem to have forgotten to add the tests for "When I click on a post, it redirects me to that post's show page." Kindly include it.

![image](https://user-images.githubusercontent.com/86318284/217517735-0ba00e02-fb10-4a5c-ae1b-0c44f3095183.png)

Reply:
In my machine rspec spec/views are not showing any error. I've attached the screenshot here.

![Screenshot (218)](https://user-images.githubusercontent.com/106825373/217534477-03469d54-e2da-45db-a759-fd987bf60b0a.png)

Reply:
Updated everything following Reviewer's review. Screenshots are also attached:


![Screenshot (219)](https://user-images.githubusercontent.com/106825373/217574137-fd0d8bf5-7623-4420-83a0-6778195129e4.png)